spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/AnnotatedServiceRegistrationBean.java

```diff
@@ -113,21 +117,32 @@ public AnnotatedServiceRegistrationBean setServiceName(@NotNull String serviceNa
     }
 
     /**
-     * Returns the decorator of the annotated service object.
+     * Returns the decorators of the annotated service object.
      */
     @NotNull
-    public Function<Service<HttpRequest, HttpResponse>,
-                    ? extends Service<HttpRequest, HttpResponse>> getDecorator() {
-        return decorator;
+    public Collection<Function<Service<HttpRequest, HttpResponse>,
+            ? extends Service<HttpRequest, HttpResponse>>> getDecorators() {
+        return decorators;
     }
 
     /**
      * Sets the decorator of the annotated service object.
+     * @deprecated Use {@link #setDecorators(Collection)} instead.
      */
+    @Deprecated
     public AnnotatedServiceRegistrationBean setDecorator(
             @NotNull Function<Service<HttpRequest, HttpResponse>,
                               ? extends Service<HttpRequest, HttpResponse>> decorator) {
-        this.decorator = decorator;
+        return setDecorators(ImmutableList.of(decorator));
+    }
+
+    /**
+     * Sets the decorators of the annotated service object.
+     */
+    public AnnotatedServiceRegistrationBean setDecorators(
+            @NotNull Collection<Function<Service<HttpRequest, HttpResponse>,
```

Comment:
How about` List<>` rather than `Collection<>` ? Due to user need to concern about decorator's order.
Also does javadoc need to explain the order?